modules/common/topology/topology.go

```diff
@@ -52,12 +55,33 @@ func EnsureTopologyRef(
 	if err != nil {
 		return topology, hash, err
 	}
+
 	// Add finalizer (if not present) to the resource consumed by the Service
 	if controllerutil.AddFinalizer(topology, fmt.Sprintf("%s-%s", h.GetFinalizer(), finalizer)) {
 		if err := h.GetClient().Update(ctx, topology); err != nil {
 			return topology, hash, err
 		}
 	}
+
+	// Set default LabelSelector on topologyConstraints if not set, similar to cluster level default:
+	// https://kubernetes.io/docs/concepts/scheduling-eviction/topology-spread-constraints/#cluster-level-default-constraints
+	topology = topology.DeepCopy()
+
+	topologyConstraints := topology.Spec.TopologySpreadConstraints
+	if topologyConstraints != nil {
+		for i := 0; i < len(*topologyConstraints); i++ {
+			current := &(*topologyConstraints)[i]
+			if current.LabelSelector == nil {
+				current.LabelSelector = defaultLabelSelector
+			}
+		}
+	}
+
+	hash, err = util.ObjectHash(topology.Spec)
```

Comment:
its probably not really expensive, but we can already get the hash on the `topologyv1.GetTopologyByName` (as it was), so we could move this inside the condition to only update the hash and calc it again when we iterate over it and there is an update.

Reply:
or could just change GetTopologyByName to not calculate the hash

Reply:
either way would work if callers do not expect GetTopologyByName to return a hash